### PR TITLE
fix: update snap packaging

### PR DESCRIPTION
**Pull request by siggiskulason**
- Add support for defining provision watcher directory
- Add auto-scan script to snap
- Add support in auto-configure script/command for providing Consul token
- Update snap base to core 20
- Update documentation
- Update metadata and versioning
- Add support for "snap set" of device-specific settings
- Update install hook
- Update make section in snapcraft.yaml
- Other minor updates

Fix https://github.com/edgexfoundry/device-rfid-llrp-go/issues/55

See also: https://github.com/edgexfoundry/device-rfid-llrp-go/pull/52 for further information.

There is an issue with setting the subnet information using "snap set", which is unrelated to this PR - see edgexfoundry/app-functions-sdk-go#1043

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rfid-llrp-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) N/A
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) N/A
  <link to docs PR>

## Testing Instructions

Install snap and run device discovery with
```
sudo snap remove edgexfoundry
sudo snap remove edgex-device-rfid-llrp
sudo snap install edgexfoundry --edge

snapcraft try
sudo snap try prime

sudo snap connect edgexfoundry:edgex-secretstore-token edgex-device-rfid-llrp:edgex-secretstore-token
sudo snap start edgex-device-rfid-llrp.device-rfid-llrp 
sudo snap connect edgex-device-rfid-llrp:network-control 
sudo snap connect edgex-device-rfid-llrp:network-observe 

CONSUL_TOKEN=$(sudo cat /var/snap/edgexfoundry/current/secrets/consul-acl-token/bootstrap_token.json | jq ".SecretID" | tr -d '"') 
echo $CONSUL_TOKEN

sudo edgex-device-rfid-llrp.auto-configure $CONSUL_TOKEN

```

I have verified that this detects my RFID reader.

### CODE REVIEW

**Comment by siggiskulason** [comment]
@lenny-intel - Besides snap-specific changes, I made a couple of changes to `driver.go` and `auto-configure.sh` which you might want to review. Thanks.

File: snap/snapcraft.yaml
Comment (farshidtz) [inline snap/snapcraft.yaml:135]: This also copies `bin/test-attribution-txt.sh` and leaves it under $SNAP. Need to include only `bin/auto-configure.sh`, or if possible just source a single file `bin/auto-configure.sh`.

**Review by farshidtz — CHANGES_REQUESTED** [review]
Thanks a lot @siggiskulason for the updates. I've added some inline comments. Apart from those, I suggest following updates:

1. Update `github.com/canonical/edgex-snap-hooks/v2` so that it includes bug fixes and better error logging (e.g. when setting wrong config option key).
2. The snap readme describes installation from 2.0 (Ireland) channel which doesn't exist. Please remove that part until we actually have a v2 channel and tag; see #63 
3. Reduce the content of "Using a content interface to set device configuration" section under snap readme to:
```
The `device-config` content interface allows another snap to seed this snap with configuration directories under `$SNAP_DATA/config/device-rfid-llrp`.

Note that the `device-config` content interface does NOT support seeding of the Secret Store Token because that file is expected at a different path.

Please refer to [edgex-config-provider](https://github.com/canonical/edgex-config-provider), for an example and further instructions.
```

File: README.md
Comment (lenny-intel) [inline README.md:42]: We are release this for Jakarta, so this needs to stay as is.

File: internal/driver/driver.go
Comment (lenny-intel) [inline internal/driver/driver.go:716]: This seems to be a missing feature in the SDK similar to `ProfilesDir` and `DevicesDir`. Please submit issue against the SDK (DS will need access to the setting) and add a TODO here to remove this once added to the SDK.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: README.md
Comment (siggiskulason) [reply README.md:42]: fixed.

File: internal/driver/driver.go
Comment (siggiskulason) [reply internal/driver/driver.go:719]: Comment added - and issue at https://github.com/edgexfoundry/device-sdk-go/issues/1100

File: snap/snapcraft.yaml
Comment (siggiskulason) [reply snap/snapcraft.yaml]: pushed an update with
```
  auto-configure:
    plugin: dump
    source: bin/
    filesets:
      auto-configure: [bin/auto-configure.sh]
    organize:
      auto-configure.sh: bin/auto-configure.sh
    stage-packages:
      - curl
 ```
which just copies this file

**Comment by siggiskulason** [comment]
> * Update `github.com/canonical/edgex-snap-hooks/v2` so that it includes bug fixes and better error logging (e.g. when setting wrong config option key).

Updated to
```
require github.com/canonical/edgex-snap-hooks/v2 v2.1.3
```

> * The snap readme describes installation from 2.0 (Ireland) channel which doesn't exist. Please remove that part until we actually have a v2 channel and tag; see #63
> * Reduce the content of "Using a content interface to set device configuration" section under snap readme to:

Updated

File: README.md
Comment (siggiskulason) [reply README.md:20]: Fixed

**Comment by siggiskulason** [comment]
@lenny-intel / @farshidtz - Thanks for the review. I have implemented all requested changes. Could you please re-review and approve if there are no other change requests? Thanks!

File: internal/driver/driver.go
Comment (lenny-intel) [reply internal/driver/driver.go:719]: Excellent. Thanks!

**Review by lenny-intel — APPROVED** [review]
LGTM, from my perspective. I'll let the other snap experts approve before merging.

File: snap/README.md
Comment (farshidtz) [inline snap/README.md:74]: Please reference the appropriate Github issue here and mention that this will work only after the issue is resolved.

The issue is now in app-functions-sdk, which isn't even used by this service: https://github.com/edgexfoundry/app-functions-sdk-go/issues/1043

File: snap/README.md
Comment (farshidtz) [inline snap/README.md:77]: The numbers make it look like the orders matter. But in fact, this is an alternative to 1. Please improve.

File: snap/README.md
Comment (farshidtz) [inline snap/README.md:151]: `service.boot-timeout` and `service.protocol` aren't supported and are already removed from some other snaps readmes.

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: snap/README.md
Comment (siggiskulason) [reply snap/README.md]: updated - please see latest commit.

File: snap/README.md
Comment (siggiskulason) [reply snap/README.md]: updated - please see latest commit.

File: snap/snapcraft.yaml
Comment (siggiskulason) [reply snap/snapcraft.yaml]: ah! thanks. updated - please see latest commit.

**Comment by siggiskulason** [comment]
@farshidtz - I've pushed a commit with your requested changes. Please re-review.

**Review by farshidtz — APPROVED** [review]
Thanks. Looks good now. 

I don't have the actual device and reviewed only the packaging and runtime of the snap.

**Review by lenny-intel — APPROVED** [review]

**Comment by siggiskulason** [comment]
Thanks @lenny-intel / @farshidtz

**Pull request merged**
